src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -125,6 +125,24 @@ public BuildScriptSnippet GenerateBashBuildScriptSnippet(BuildScriptGeneratorCon
             }
 
             bool pruneDevDependencies = ShouldPruneDevDependencies(context);
+            bool appInsightDependency = IsPackageDependencyExist(packageJson, NodeConstants.NodeAppInsightPackageName);
+            string appInsightInjectCommand = string.Empty;
+
+            // node_options is only supported in version 8.0.0 or newer and in 6.12.0
+            // so we will be able to set up app-insight only when node version is 6.12.0 or 8.0.0 or newer
+            if (!appInsightDependency
+                && !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.AppInsightKey))
+                && ( SemanticVersionResolver.CompareVersion(context.NodeVersion, "8.0.0") >= 0
+                || SemanticVersionResolver.CompareVersion(context.NodeVersion, "6.12.0") == 0
+                || SemanticVersionResolver.CompareVersion(context.LanguageVersion, "8.0.0") >= 0
+                || SemanticVersionResolver.CompareVersion(context.LanguageVersion, "6.12.0") == 0))
+            {
+                appInsightInjectCommand =
+                    string.Concat(NodeConstants.NpmPackageInstallCommand, " --save ", NodeConstants.NodeAppInsightPackageName);
+                buildProperties[NodeConstants.OryxNodeOptions] = true.ToString();
```

Comment:
Can you use another key name other than `OryxNodeOptions` as it does not indicate what that key is for? something like `loadAppInsightsModule` or something